### ISSUE TITLE
Unexpected "recursive value needs type" error

### ISSUE THREAD

**Issue by scabug** [scala/bug#5091]
With scala-2.10.0.r25850-b20111019023417, compiling:
```scala
// RecursiveValueNeedsType.scala
object RecursiveValueNeedsType {

  def foo(param: String) = 42
  def bar(n: Int) = 42

  {
    val xxx = foo(param = null)
    val param = bar(xxx)
  }
  
}
```

Gives the error:
```
RecursiveValueNeedsType.scala:8: error: recursive value xxx needs type
    val param = bar(xxx)
```

**Comment by Abrasha** [comment]
The same here: https://stackoverflow.com/questions/45814226/scala-named-argument-recursive-call/45816574#45816574

**Comment by xuwei-k** [comment]
fixed since Scala 2.13.0-M4

https://github.com/scala/scala/commit/a992c8352b9f309ec0a3d807425786dc21dd6284#diff-3828abb025ee93aa4f0b1d654200e5c9
